File: sdk/js/src/parse/defaults.ts
import { ABSENT, type Absent } from "../schemas/base.js";

/**
 * Apply a default value if the input is absent.
 * Returns the default value if input is absent, otherwise returns the input as-is.
 */
export function applyDefault<T>(
  input: unknown,
  defaultValue: T | Absent
): unknown {
  if ((input === undefined || input === ABSENT) && defaultValue !== ABSENT) {
    return defaultValue;
  }
  return input;
}

File: sdk/rust/src/parse/defaults.rs
use serde_json::Value;

/// Apply a default value if the input is absent (None/undefined).
/// Returns the default value if input is None, otherwise the input.
pub fn apply_default(input: Option<&Value>, default: &Value) -> Value {
    match input {
        Some(v) => v.clone(),
        None => default.clone(),
    }
}

/// Check if a value should be treated as absent for default application.
/// Note: null is NOT treated as absent per the spec.
pub fn is_absent(input: Option<&Value>) -> bool {
    input.is_none()
}
